casanovo/denovo/model.py

```diff
@@ -989,6 +990,208 @@ def configure_optimizers(
         return [optimizer], {"scheduler": lr_scheduler, "interval": "step"}
 
 
+class DbSpec2Pep(Spec2Pep):
+    """
+    Inherits Spec2Pep
+
+    Hijacks teacher-forcing implemented in Spec2Pep and
+    uses it to predict scores between a spectra and associated peptide.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def predict_step(self, batch, *args):
+        """
+        A single prediction step for Casanovo-DB
+
+        Parameters
+        ----------
+        batch : Tuple[torch.Tensor, torch.Tensor, torch.Tensor]
```

Comment:
Currently in `full_db_search`, this is correct because the candidate peptides are calculated from the precursor and are not passed in with the batch (this may be changed if I move this batching logic to the dataloader)